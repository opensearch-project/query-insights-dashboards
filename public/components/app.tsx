/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { Route } from 'react-router-dom';
import { DataSourceManagementPluginSetup } from 'src/plugins/data_source_management/public';
import TopNQueries from '../pages/TopNQueries/TopNQueries';
import { AppMountParameters, CoreStart } from '../../../../src/core/public';
import { QueryInsightsDashboardsPluginStartDependencies } from '../types';

export const QueryInsightsDashboardsApp = ({
  core,
  depsStart,
  params,
  dataSourceManagement,
}: {
  core: CoreStart;
  depsStart: QueryInsightsDashboardsPluginStartDependencies;
  params: AppMountParameters;
  dataSourceManagement?: DataSourceManagementPluginSetup;
}) => {
  return (
    <Route
      render={() => (
        <TopNQueries
          core={core}
          depsStart={depsStart}
          params={params}
          dataSourceManagement={dataSourceManagement}
        />
      )}
    />
  );
};
