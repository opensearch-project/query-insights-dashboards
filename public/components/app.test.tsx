/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import { QueryInsightsDashboardsApp } from './app';

describe('<QueryInsightsDashboardsApp /> spec', () => {
  it('renders the component', () => {
    const mockHttpStart = {
      basePath: {
        serverBasePath: '/app/opensearch-dashboards',
      },
    };

    const { container } = render(
      <QueryInsightsDashboardsApp
        basename="/"
        http={mockHttpStart as any}
        navigation={
          {
            ui: { TopNavMenu: () => null },
          } as any
        }
        notifications={{} as any}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
