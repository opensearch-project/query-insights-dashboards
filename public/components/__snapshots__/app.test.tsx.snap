// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueryInsightsDashboardsApp /> spec renders the component 1`] = `
<div>
  <div
    class="euiPage euiPage--paddingMedium euiPage--grow euiPage--restrictWidth-custom"
    style="max-width: 1000px;"
  >
    <main
      class="euiPageBody euiPageBody--borderRadiusNone"
    >
      <header
        class="euiPageHeader euiPageHeader--responsive euiPageHeader--center"
      >
        <h1
          class="euiTitle euiTitle--large"
        >
          query-insights-dashboards
        </h1>
      </header>
      <div
        class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageContent"
        role="main"
      >
        <div
          class="euiPageContentHeader euiPageContentHeader--responsive"
        >
          <h2
            class="euiTitle euiTitle--medium"
          >
            Congratulations, you have successfully created a new OpenSearch Dashboards Plugin!
          </h2>
        </div>
        <div
          class="euiPageContentBody"
        >
          <div
            class="euiText euiText--medium"
          >
            <p>
              Look through the generated code and check out the plugin development documentation.
            </p>
            <hr
              class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
            />
            <p>
              Last timestamp: Unknown
            </p>
            <button
              class="euiButton euiButton--primary euiButton--small"
              type="primary"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Get data
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
